dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -237,15 +241,50 @@ impl<'a, 'tcx: 'a> CollectFunctionInstrumentationPoints<'a, 'tcx> {
     }
 
     fn should_instrument(&self, local: Local, context: PlaceContext) -> bool {
-        let is_unsafe_ptr = self.body.local_decls[local].ty.is_unsafe_ptr();
+        let is_unsafe_ptr = self.is_shared_region_or_unsafe_ptr(self.body.local_decls[local].ty);
         is_unsafe_ptr && context.is_use()
     }
 
+    fn is_unsafe_ptr(&self, ty: &TyS<'tcx>) -> bool {
+        ty.is_unsafe_ptr()
+    }
+
+    fn is_shared_region_or_unsafe_ptr(&self, ty: &TyS<'tcx>) -> bool {
```

Comment:
I would call this `is_shared_or_unsafe_ptr`; `region` doesn't disambiguate anything and `shared ptrs` are a perfectly valid thing to call `&T` that does still distinguish from `&mut T`.